src/main/java/org/spongepowered/api/text/title/TitleBuilder.java

```diff
@@ -24,91 +24,311 @@
  */
 package org.spongepowered.api.text.title;
 
-import org.spongepowered.api.text.message.Message;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Optional;
+import org.spongepowered.api.text.Text;
+
+import javax.annotation.Nullable;
 
 /**
- * Represents a builder interface to create immutable {@link Title}
- * configurations.
+ * Represents a builder class to create immutable {@link Title} configurations.
+ *
+ * @see Title
  */
-public interface TitleBuilder {
+public class TitleBuilder {
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  adam@eq2.serveftp.net
    Domain eq2.co.uk has exceeded the max emails per hour (68/60 (113%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o4.sgmail.github.com ([192.254.112.99]:7138)
	by web3.cov.yesit.co.uk with esmtps (TLSv1.2:DHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.84)
	(envelope-from <bounces+848413-d0b0-adam=eq2.co.uk@sgmail.github.com>)
	id 1YNeaw-0003Zf-AE
	for adam@eq2.co.uk; Tue, 17 Feb 2015 09:37:30 +0000
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=pwNQZ4Nh4cDK8dS9M2fEdRAZP9I=; b=KVr0f0Cs8fmH/ia6
	RXx1IIH8OQj2HAfuZs3PYvC0LZ/9UF0nhMIpbCP6qGNte674akll/+j/KN3Prn9+
	l/lJGsq1jBEnykNf/gU/LXwR4TZK6Gfz94uoUF/Ab9Imxe/8qIoLkMOv8qPQE4u8
	D2pk6TrZbC7FvQtNq3XA7D5uFYA=
Received: by filter0215p1mdw1.sendgrid.net with SMTP id filter0215p1mdw1.32028.54E30BD816
        2015-02-17 09:37:28.830925869 +0000 UTC
Received: from github-smtp2b-ext-cp1-prd.iad.github.net (github-smtp2b-ext-cp1-prd.iad.github.net [192.30.253.17])
	by ismtpd-025 (SG) with ESMTP id 14b96e646a7.2f88.7399ef
	for <adam@eq2.co.uk>; Tue, 17 Feb 2015 09:37:28 +0000 (UTC)
Date: Tue, 17 Feb 2015 01:37:28 -0800
From: gratimax <notifications@github.com>
Reply-To: SpongePowered/SpongeAPI <reply@reply.github.com>
To: SpongePowered/SpongeAPI <SpongeAPI@noreply.github.com>
Cc: Mumfrey <adam@eq2.co.uk>
Message-ID: <SpongePowered/SpongeAPI/pull/425/r24803140@github.com>
In-Reply-To: <SpongePowered/SpongeAPI/pull/425@github.com>
References: <SpongePowered/SpongeAPI/pull/425@github.com>
Subject: Re: [SpongeAPI] [WIP] Text API 2 (#425)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_54e30bd88a071_32883fb39ed2729c14856f9";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: gratimax
X-GitHub-Recipient: Mumfrey
X-GitHub-Reason: comment
List-ID: SpongePowered/SpongeAPI <SpongeAPI.SpongePowered.github.com>
List-Archive: https://github.com/SpongePowered/SpongeAPI
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+001180bfe517ae5986187c425ee63f1a1746b7a86e5c7af592cf0000000110facdd892a170ce017a7744@reply.github.com>,
 <https://github.com/notifications/unsubscribe/ABGAv9VSUb5E316UQFRwyrkQN-WDOGUIks5nswNYgaJpZM4DZr_C>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: adam@eq2.co.uk
X-SG-EID: VzkE0stiinYMPYWPWpnZ4OvH/Ru8MrIuYzb7JqhnFX8ogkFD+CRpc5vv7MwMf0pECg/KHNXGndNUh1
 YmxVv1ptj/fyEtsdzPG5HpBamKCPeBL9cLoD9VBA0Mzj8ZZVkJ0kqKNpKDHHrk/5Vtz2D1HVQq4Nyf
 PzewVWGFyTA63/s=

----==_mimepart_54e30bd88a071_32883fb39ed2729c14856f9
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

>   */
> -public interface TitleBuilder {
> +public class TitleBuilder {

What's with the `public final` methods?

---
Reply to this email directly or view it on GitHub:
https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803140
----==_mimepart_54e30bd88a071_32883fb39ed2729c14856f9
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/SpongePowered/SpongeAPI/pull/425#discussion_r24803140">src/main/java/org/spongepowered/api/text/title/TitleBuilder.java</a>:</p>
<pre style='color:#555'>&gt;   */
&gt; -public interface TitleBuilder {
&gt; +public class TitleBuilder {
</pre>
<p>What's with the <code>public final</code> methods?</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803140">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/ABGAv4XGTSRADIli0g4jEWB_3usdAfknks5nswNYgaJpZM4DZr_C.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
  <div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
    <link itemprop="url" href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803140"></link>
    <meta itemprop="name" content="View Pull Request"></meta>
  </div>
  <meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_54e30bd88a071_32883fb39ed2729c14856f9--